src/components/header/headerElement.js

```diff
@@ -20,7 +21,13 @@ const HeaderElement = props => (
         theme={styles}
       >
         <MenuItem caption="Register second passphrase" />
-        <MenuItem caption="Register as delegate" />
+        <MenuItem caption="Register as delegate"
+          className={(props.account.isDelegate) ? styles.hidden : ''}
```

Comment:
I see the PR is in progress, but I just noticed this one thing. IMO more elegant way to conditionally hide an element in react is the ternary operator - `{shouldShow ? <Element /> : null }`.